src/cli/ask.rs

```diff
@@ -82,46 +80,51 @@ pub fn ask(
     let span = call.head;
 
     let question: String = call.req(engine_state, stack, 0)?;
-    let mut context: Vec<String> = Vec::new();
-    match input.into_value(span) {
-        Value::List {
-            vals,
-            internal_span: span,
-        } => {
-            for v in vals {
-                let rec = match v.as_record() {
-                    Ok(r) => r,
-                    Err(e) => {
+    let context: Vec<String> = match call.opt(engine_state, stack, 1)? {
+        Some(ctx) => ctx,
+        None => match input.into_value(span) {
+            Value::List {
+                vals,
+                internal_span: span,
+            } => {
+                let mut ctx: Vec<String> = Vec::new();
+                for v in vals {
+                    let rec = match v.as_record() {
+                        Ok(r) => r,
+                        Err(e) => {
+                            return Err(ShellError::GenericError {
+                                error: "Context must be a nushell table".to_string(),
+                                msg: "".to_string(),
+                                span: Some(span),
+                                help: None,
+                                inner: vec![e],
+                            });
+                        }
+                    };
+
+                    if rec.columns().len() > 1 {
                         return Err(ShellError::GenericError {
-                            error: "Context must be a nushell table".to_string(),
+                            error: "Use 'select' to choose a single column to pipe to 'ask'"
+                                .to_string(),
                             msg: "".to_string(),
                             span: Some(span),
                             help: None,
-                            inner: vec![e],
+                            inner: vec![],
                         });
                     }
-                };
 
-                if rec.columns().len() > 1 {
-                    return Err(ShellError::GenericError {
-                        error: "Use 'select' to choose a single column to pipe to 'ask'"
-                            .to_string(),
-                        msg: "".to_string(),
-                        span: Some(span),
-                        help: None,
-                        inner: vec![],
-                    });
+                    let ctx_string = rec.get_index(0).unwrap().1.as_str()?;
```

Comment:
Is it possible for rec to be `{}`?

Reply:
It could be, have added handling of that case